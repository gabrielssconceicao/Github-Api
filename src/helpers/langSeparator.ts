import type { IRepos } from '@/types/repos';
type ILang = { name: string; count: number };
export const langSeparator = (repos: IRepos[]): ILang[] => {
  const langs: ILang[] = [];
  repos.forEach((repo) => {
    const lang = repo.language;
    const index = langs.findIndex((l) => l.name === lang);
    console.log(index);
    if (index === -1) {
      langs.push({ name: lang, count: 1 });
    } else {
      langs[index].count += 1;
    }
    console.log(langs);
  });
  return langs;
};
