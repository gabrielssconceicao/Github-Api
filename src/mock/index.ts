import { type IUser } from '@/types/user';
import { type IRepos } from '@/types/repos';

export const user: IUser = {
  name: 'Teste User',
  login: 'testeuser',
  avatar_url: 'https://avatars.githubusercontent.com/u/10590076?v=4',
  followers: 0,
  following: 5,
  location: 'Brazil',
  html_url: 'https://github.com/testeuser',
  bio: 'lorem ipsum dolor sit amet',
  public_repos: 5,
};
// JS - 3
// HTML - 2
// python - 1
// Ruby - 1
//Java - 1
//C++ - 1
//Typescript - 1
export const repos: IRepos[] = [
  {
    name: 'Teste Repo 1',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 2',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'HTML',
  },
  {
    name: 'Teste Repo 3',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Python',
  },
  {
    name: 'Teste Repo 4',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 5',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Javascript',
  },
  {
    name: 'Teste Repo 6',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'HTML',
  },
  {
    name: 'Teste Repo 7',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Ruby',
  },
  {
    name: 'Teste Repo 8',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Java',
  },
  {
    name: 'Teste Repo 9',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'C++',
  },
  {
    name: 'Teste Repo 10',
    description: 'lorem ipsum dolor sit amet',
    html_url: 'https://github.com/testeuser/testerepo',
    language: 'Typescript',
  },
];
