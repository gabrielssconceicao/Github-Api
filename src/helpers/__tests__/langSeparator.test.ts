import { describe, it, expect } from 'vitest';
import { langSeparator } from '../langSeparator';
import { repos } from '@/mock';
describe('langSeparator', () => {
  it('should separate langs from repos and count them', () => {
    const result = langSeparator(repos);
    const expected = [
      { name: 'Javascript', count: 3 },
      { name: 'HTML', count: 2 },
      { name: 'Python', count: 1 },
      { name: 'Ruby', count: 1 },
      { name: 'Java', count: 1 },
      { name: 'C++', count: 1 },
      { name: 'Typescript', count: 1 },
    ];
    expect(result).toEqual(expected);
  });

  it('should match snapshot', () => {
    const result = langSeparator(repos);
    expect(result).toMatchSnapshot();
  });
});
