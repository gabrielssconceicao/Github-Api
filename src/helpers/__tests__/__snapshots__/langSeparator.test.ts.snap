// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`langSeparator > should match snapshot 1`] = `
[
  {
    "count": 3,
    "name": "Javascript",
  },
  {
    "count": 2,
    "name": "HTML",
  },
  {
    "count": 1,
    "name": "Python",
  },
  {
    "count": 1,
    "name": "Ruby",
  },
  {
    "count": 1,
    "name": "Java",
  },
  {
    "count": 1,
    "name": "C++",
  },
  {
    "count": 1,
    "name": "Typescript",
  },
]
`;
